src/aiida/manage/configuration/settings.py

```diff
@@ -66,45 +66,96 @@ def create_instance_directories() -> None:
 
-    elif environment_variable := os.environ.get(DEFAULT_AIIDA_PATH_VARIABLE):
-        # Loop over all the paths in the `AIIDA_PATH` variable to see if any of them contain a configuration folder
-        for base_dir_path in [path for path in environment_variable.split(':') if path]:
-            AIIDA_CONFIG_FOLDER = pathlib.Path(base_dir_path).expanduser()
+    return dirpath_config
 
-            # Only add the base config directory name to the base path if it does not already do so
-            # Someone might already include it in the environment variable. e.g.: AIIDA_PATH=/home/some/path/.aiida
-            if AIIDA_CONFIG_FOLDER.name != DEFAULT_CONFIG_DIR_NAME:
-                AIIDA_CONFIG_FOLDER = AIIDA_CONFIG_FOLDER / DEFAULT_CONFIG_DIR_NAME
 
-            # If the directory exists, we leave it set and break the loop
-            if AIIDA_CONFIG_FOLDER.is_dir():
-                break
-    else:
-        # The `AIIDA_PATH` variable is not set so use the default path and try to create it if it does not exist
-        AIIDA_CONFIG_FOLDER = pathlib.Path(DEFAULT_AIIDA_PATH).expanduser() / DEFAULT_CONFIG_DIR_NAME
+def get_configuration_directory_from_envvar(environment_variable: str) -> pathlib.Path:
+    """Return the path of a config directory from.
+
+    The environment variable should be a colon separated string of filepaths that either point directly to a config
+    directory or a path that contains a config directory. The first match is returned. If no existing config directory
+    is found, the last path in the environment variable is used.
+
+    :returns: The path of the configuration directory.
+    """
+    # Loop over all the paths in the ``AIIDA_PATH`` variable to see if any of them contain a configuration folder
+    for base_dir_path in [path for path in environment_variable.split(':') if path]:
+        dirpath_config = pathlib.Path(base_dir_path).expanduser()
+
+        # Only add the base config directory name to the base path if it does not already do so
+        # Someone might already include it in the environment variable. e.g.: ``AIIDA_PATH=/home/some/path/.aiida``
+        if dirpath_config.name != DEFAULT_CONFIG_DIR_NAME:
+            dirpath_config = dirpath_config / DEFAULT_CONFIG_DIR_NAME
+
+        # If the directory exists, we leave it set and break the loop
+        if dirpath_config.is_dir():
+            break
+
+    return dirpath_config
+
+
+def get_configuration_directory_from_cwd() -> pathlib.Path | None:
+    """Return the path of the first occurrence of a config directory in the hierarchy of the current working directory.
+
+    :returns: The path of an existing config directory in the hierarcy of the current working directory or ``None`` if
+        no such directory exists.
+    """
+    # Start at the current working directory checking for existing configuration directories, moving up the hierarchy
+    # until the first directory is found or the root directory is hit.
```

Comment:
nit: redundant given the docstring above?
```suggestion
    # Start at the current working directory checking for existing configuration directories, moving up the hierarchy
    # until the first directory is found or the root directory is hit.
```